solutions/gold/usaco-384.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: usaco-384
+source: USACO Gold 2014 January
+title: Ski Course Rating
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_skilevel.html)
+
+## Implementation
+
+<LanguageSection>
+<CPPSection>
+
+Interestingly, the following code passes all test cases but won't pass if $T = 1$; you'd have to make a separate edge case for that.
+
+**Time Complexity:** $\mathcal{O}(N^2 \log N^2)$
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+struct Edge {
+	int u, v, w;
+};
+
+struct DSU {
+	vector<int> p, sz, cnt_start;
```

Comment:
rename `cnt_start` to `unprocessed_starts` bc it seeems that's what it actually contains

Reply:
i think my comment there is good enough isnt it?

Reply:
why add a comment when you could just add it into the name itself

Reply:
im getting karma